concent_api/core/payments/sci_backend.py

```diff
@@ -0,0 +1,140 @@
+import math
+from decimal        import Decimal
+from django.conf    import settings
+from golem_messages import message
+from golem_sci      import implementation
+
+
+def sum_payments(list_of_payments: list = None):
+    assert isinstance(list_of_payments, list) and not None
+
+    return sum([item.amount for item in list_of_payments])
+
+
+def sum_subtask_price(subtask_results_accepted_list: list = None):
+    assert isinstance(subtask_results_accepted_list, list) and not None
+
+    return sum([subtask_results_accepted.task_to_compute.subtask_price for subtask_results_accepted in subtask_results_accepted_list])
+
+
+def sum_amount_price_for_provider(
+    list_of_forced_payments:        list = None,
+    list_of_payments:               list = None,
+    subtask_results_accepted_list:  list = None,
+):
+    assert isinstance(list_of_payments,                 list) and not None
+    assert isinstance(list_of_forced_payments,          list) and not None
+    assert isinstance(subtask_results_accepted_list,    list) and not None
+
+    force_payments_price    = sum_payments(list_of_forced_payments)
+    payments_price          = sum_payments(list_of_payments)
+    subtasks_price          = sum_subtask_price(subtask_results_accepted_list)
+
+    amount_paid     = payments_price + force_payments_price
+    amount_pending  = subtasks_price - amount_paid
+
+    return (amount_paid, amount_pending)
+
+
+def get_clients_eth_accounts(task_to_compute: message.tasks.TaskToCompute):
+    assert isinstance(task_to_compute, message.tasks.TaskToCompute) and not None
+
+    requestor_eth_address   = task_to_compute.requestor_ethereum_public_key
+    provider_eth_address    = task_to_compute.provider_ethereum_public_key
+
+    return (requestor_eth_address, provider_eth_address)
+
+
+def get_number_of_eth_block(concent_rpc: implementation.SCIImplementation):
+    assert isinstance(concent_rpc, implementation.SCIImplementation) and not None
```

Comment:
` and not None` in not needed, if it would be an instance, it can't be `None`.

Reply:
Done